lib/flop_phoenix.ex

```diff
@@ -263,6 +263,32 @@ defmodule Flop.Phoenix do
   which will be ignored in `.eex` templates. When used in LiveView templates,
   you will need to handle the new params in the `handle_params/3` callback of
   your LiveView module.
+
+  ### Event Based Pagination and Sorting
+  To make `Flop.Phoenix` use event based pagination and sorting you must set the
+  `:event_name` option on the pagination and table generators. This will generate
+  an `<a>` tag with `phx-click` and `phx-value` attributes set. You will need to handle
+  the event using `handle_event/3` callbacks of your LiveView module using the
+  event name you set in options.
+
+      def handle_event("pagiante_pets", %{"page" => page}, socket) do
+        flop = Flop.Phoenix.ensure_page_based_params(socket.assigns.meta.flop)
```

Comment:
I feel like we shouldn't burden the user with cleaning up the parameters.
Maybe we can pass all the needed params as `phx_value_*` options, or something like that.

Reply:
Alternatively, we could add a helper function like `set_page`, similar to `push_order`.

Reply:
I tried setting phx value-* initially but had some issues with that. For example phx_value_page_size will arrive as page-size and the user will still have to pattern match the params to create his own Flop struct using page_size. Another example was sending a list of atoms for sorting was giving me a hard time. So my thought was that the user should be responsible for managing the Meta data and all we send over as event params are the necessary changes. I like the idea of the set_page similar to push_order. makes it easier to understand and is consistent.